integrations/mcp/src/haystack_integrations/tools/mcp/mcp_toolset.py

```diff
@@ -0,0 +1,232 @@
+import logging
+from typing import Any
+
+import httpx
+from exceptiongroup import ExceptionGroup
```

Comment:
Could we add `httpx` as explicit dependencies to the pyproject.toml? It looks like right now we are relying on this being imported by mcp or haystack.